### ISSUE TITLE
Tar failure due to device being busy in snapshot tests

### ISSUE THREAD

**Issue by ZackaryWelch** [zfsonfreebsd/ZoF#34]
Snapshot_002 and snapshot_006 both fail due to a tar command. In snapshot_002, a bunch of files are create and then put into an archive, but the command fails as follows:
```
NOTE: Create a tarball from /tests/testdir contents...
tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: lseek(SEEK_HOLE) failed: Device busy tar: Error exit delayed from previous errors.
ERROR: tar cf /tests/testdir/tarball.original.tar file1 file10 file11 file12 file13 file14 file15 file16 file17 file18 file19 file2 file20 file3 file4 file5 file6 file7 file8 file9 exited 1
N
```
Snapshot_006 has a very similar error.

**Comment by ZackaryWelch** [comment]
I cannot replicate this error outside of the test. Any idea what the lseek is from, or why this might be happening? Maybe other tests are running into this. The 'tar: lseek(SEEK_HOLE) fail: Device busy' is duplicated for each file.

**Comment by ZackaryWelch** [comment]
Setup
In snapshot_002:
```
file_write -o create -f /var/tmp/testdir/file1  -b 8192 -c 20 -d 0
file_write -o create -f /var/tmp/testdir/file2  -b 8192 -c 20 -d 0
file_write -o create -f /var/tmp/testdir/file3  -b 8192 -c 20 -d 0
file_write -o create -f /var/tmp/testdir/file4  -b 8192 -c 20 -d 0
file_write -o create -f /var/tmp/testdir/file5  -b 8192 -c 20 -d 0
file_write -o create -f /var/tmp/testdir/file6  -b 8192 -c 20 -d 0
file_write -o create -f /var/tmp/testdir/file7  -b 8192 -c 20 -d 0
file_write -o create -f /var/tmp/testdir/file8  -b 8192 -c 20 -d 0
cd /var/tmp/testdir
tar cf /var/tmp/testdir/tarball.original.tar file*
```
I was unable to verify this outside of the test. Even when stopping the test before the tarball was created to make one of my own. So, good luck. The original test creates 21 files, but I found the error occurs with any number.

**Comment by mattmacy** [comment]
@kmoore134 can you please figure out how to replicate this failure outside of the test?

**Comment by kmoore134** [comment]
@mattmacy @ZackaryWelch Ok, its 100% reproducible using the instructions Zack provided. The issue was that he used /var/tmp/testdir as the file location, which is on the UFS partition, not the ZFS pool. If I test it here with my ZoL installation it fails immediately:

```                                                                        
[kris@avenger] ~/trueos/trueos-ports/sysutils/zol/work/ZoF-e10a5fb/tests/zfs-tests/cmd/file_write% sudo ./file_write -o create -f /root/file1 -b 8192 -c 20 -d 0    
% sudo tar cvf /root/backup.tar /root/file1                                                                                                         
tar: lseek(SEEK_HOLE) failed: Device busy
tar: Error exit delayed from previous errors.
```

Note that this is a ZoL on root installation, so /root is ZFS in this case.

**Comment by ZackaryWelch** [comment]
Thanks @kmoore134, I was unable to replicate it outside the test earlier. I think I wrote the wrong testdir down in that example.

**Comment by mattmacy** [comment]
I'll look at it now

**Comment by mattmacy** [comment]
fixed by e309a86f4d518401995f7a7954e1d010c5ee1d30
